### ISSUE TITLE
triple slash in source file has unexpected global effect

### ISSUE THREAD

**Issue by yw662** [microsoft/TypeScript#39100]
**TypeScript Version:**  3.9.5

**Search Terms:**
triple slash
**Code**

The code has 3 parts, a `tsconfig.json`, and 2 `.ts` source code, let's call them `dom.ts` and `webworker.ts`.

`tsconfig.json`:
```json
{
    "compilerOptions": {
        "module": "ESNext",
        "target": "ESNext",
        "lib": [
            "ESNext"
        ],
        "outDir": "build",
        "allowJs": false,
        "strict": true,
        "alwaysStrict": true,
    },
    "files": [
        "src/dom.ts",
        "src/webworker.ts"
    ],
}
```

`dom.ts`
```ts
/// <reference lib="dom" />
```

`webworker.ts`
```ts
/// <reference lib="WebWorker" />
```

**Expected behavior:**
It should pass and compile.
**Actual behavior:**
Although it generates the target files correctly, at least for this demo, 26 errors are reported in `lib.dom.d.ts` and `lib.webworker.d.ts`.

**Playground Link:** 
Seems that playground cannot handle this.

**Related Issues:** 
Maybe not, but #33901

**Comment by RyanCavanaugh** [comment]
The lib files add globals to the global scope; the global scope is shared by all files by virtue of being global. The recommended way forward here would be to have separate tsconfig files (one for your webworker code, one for your dom code)
